Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -40,6 +43,32 @@ The following is a sample attestation result:
 }
 ```
 
+**ctsProfileMatch Vs basicIntegrity**
+
+The SafetyNet Attestation API initially provided a single value called basicIntegrity to help developers determine the integrity of a device. As the API evolved, Google introduced a new, stricter check whose results appear in a value called ctsProfileMatch, which allows developers to more finely evaluate the devices on which their app is running.
+
+In broad terms, basicIntegrity gives you a signal about the general integrity of the device and its API. Rooted devices fail basicIntegrity, as do emulators, virtual devices, and devices with signs of tampering, such as API hooks.
+
+On the other hand, ctsProfileMatch gives you a much stricter signal about the compatibility of the device. Only unmodified devices that have been certified by Google can pass ctsProfileMatch. Devices that will fail ctsProfileMatch include the following:
+
+- Devices that fail basicIntegrity
+- Devices with an unlocked bootloader
+- Devices with a custom system image (custom ROM)
+- Devices for which the manufactured didn't apply for, or pass, Google certification
+- Devices with a system image built directly from the Android Open Source Program source files
+- Devices with a system image distributed as part of a beta or developer preview program (including the Android Beta Program)
+
+**Do's and Don'ts of** `SafetyNetApi.attest`
+
+-  Create a large (16 bytes or longer) random number on your server using a cryptographically-secure random function so that a malicious user can not reuse a successful attestation result in place of an unsuccessful result

```

Comment:
Is this a do or a don't ;-), if you know what it is about, you know this is a do, but it is better to change the caption from `Do's and Don'ts` into: `Recommendations when using` ...

Reply:
Agree